### ISSUE TITLE
Authentication and Decryption Weirdness

### ISSUE THREAD

**Issue by Lorb1** [vector-im/element-web#17048]
I experience issues, that i can't really narrow down.
I am using a Windows 10 PC and have enabled a second passphrase as cross signing verification.
I have an private room with two other members and own chatrooms for each person.
Encryption is enabled in all rooms.

With one of the members, who uses an IPhone, i have close to no problem. The only thing weird is, that:
1. In the group- and the private room his messages are displayed with a grey shield infront of them, stating:

_"The authenticity of this encrypted message can't be guaranteed on this device."_

We both have each other verified in the provate and the group room.

2. Messages sent in the private room seem to be normal at first glance. Green shield next to the textfield and next to the users.
But when logging out and logging back in, my sent messages are displayed with a red shield next to them, stating:

_"Encrypted by a deleted session"_

With the other user, who is using three devices (Andoird Phone, Chromebook and Windows 7 PC, that is not really beeing used) on the other hand there are much more severe issues:

In the private room:

He is displayed with a red shield stating:

_"This user has not verified all of their sessions."_

i am beeing displayed with a green shield:

"You have verified this user. This user has verified all of their sessions."

He has enabled a second passphrase for validation. Also we both have verified each other with every of his devices. Even though the process had succes every time. After logging out and back in, he is still displayed as not beeing verified and redoing the verifying procedure goes exactly the same.

The text input shows a red shield next to it:

_"Someone is using an unkown session"_

Old messages of mine have a red shield stating:

_"Encrypted by a deleted session"_
also this error only appears after logging out an back in.

Messages he wrote to me during a sessions are beeing displayed with a red shield stating:

_"Encrypted by an unverified session"_

Messages he wrote to me while i was offline are not decryptable, with following error:
 ** Unable to decrypt: The sender's device has not sent us the keys for this message. ** 
_Re-request encryption keys from your other sessions_

Clicking the _Re-request encryption keys from your other sessions_ does nothing.

In the group room:

The member with the IPhone and i are displayed with a green shield:

_"You have verified this user. This user has verified all of their sessions."_

The user i am currently talking about has a red shield infront of him stating:

_"This user has not verified all of their sessions."_

There's a red shield next to the text input and the room icon stating:

_"Someone is using an unkown session"_

My messages are displayed with the error:

_"Encrypted by a deleted session"_
Aswell, only after logging out and back in.

Same with the grey shield _"The authenticity of this encrypted message can't be guaranteed on this device."_ of the IPhone user.

The messages of the user with the three devices ahve a red shield infornt of them stating:

_"Encrypted by an unverified session"_
Also only after logging out and back in.

When i, or the IPhone are offline and the Threedevice writes a message, we both are unables to decrypt them.

 ** Unable to decrypt: The sender's device has not sent us the keys for this message. ** 
_Re-request encryption keys from your other sessions_

Clicking on the  _Re-request encryption keys from your other sessions_ again does nothing for me. The IPhone user on the other hand gets following message:

"**Request sent** Please open Element on an other device, that can encrypt the message, so it can send the the keys to the other session"

Remember this user does not have any other device for element.

### Steps to reproduce

I have no idea.

As i have stated, i am using Windows 10 and element as a desktop app. The IPhone user uses Element as an app.
The Three Device user uses Element as a desktop app, smartphone app, an chromebook app.
So noone uses element through a browser.

**Comment by kittykat** [comment]
> 1. In the group- and the private room his messages are displayed with a grey shield infront of them, stating:
> 
> _"The authenticity of this encrypted message can't be guaranteed on this device."_
> 
> We both have each other verified in the provate and the group room.
> 
> 2. Messages sent in the private room seem to be normal at first glance. Green shield next to the textfield and next to the users.
>    But when logging out and logging back in, my sent messages are displayed with a red shield next to them, stating:
> 
> _"Encrypted by a deleted session"_

When you log out, you are effectively deleting that session so that message is correct. Have you verified the new session against the deleted one?

> With the other user, who is using three devices (Andoird Phone, Chromebook and Windows 7 PC, that is not really beeing used) on the other hand there are much more severe issues:
> 
> In the private room:
> 
> He is displayed with a red shield stating:
> 
> _"This user has not verified all of their sessions."_
> 
> i am beeing displayed with a green shield:
> 
> "You have verified this user. This user has verified all of their sessions."

You/he needs to check if all of his devices are verified and whether old sessions are deleted. You can do this in the right panel on web/desktop, when you look at his profile:

![Screenshot from 2021-12-23 11-21-01](https://user-images.githubusercontent.com/51663/147233533-c3d37833-4132-49c9-8105-3cdf8046f6f2.png)

> Messages he wrote to me while i was offline are not decryptable, with following error: ** Unable to decrypt: The sender's device has not sent us the keys for this message. ** _Re-request encryption keys from your other sessions_

This can be a symptom of not having encryption set up correctly. It may be worth re-setting in "Security & Privacy".

If you are still experiencing problems after that, and after confirming that all three of you have only verified devices, then it's worth triggering the decryption issue and submitting rageshakes from the sender and receiver devices against one new issue which describes only that problem and uses one of our bug templates as this would flag the issue for triage.

I'm going to close this issue for now as it cannot be investigated further without logs/rageshakes from all devices involved.

If you have any support questions, I would recommend asking in [#element-web:matrix.org for the desktop app](https://matrix.to/#/#element-web:matrix.org), [#element-ios:matrix.org](https://matrix.to/#/#element-ios:matrix.org) or [#element-android:matrix.org](https://matrix.to/#/#element-android:matrix.org)
